resolver/ede_resolver_test.go

```diff
@@ -81,8 +82,17 @@ var _ = Describe("EdeResolver", func() {
 			}
 		})
 
-		extractFirstOptRecord := func(e []dns.RR) []dns.EDNS0 {
-			return e[0].(*dns.OPT).Option
+		extractEdeOption := func(res *Response) dns.EDNS0_EDE {
+			opt := res.Res.IsEdns0()
+			if opt != nil {
+				for _, o := range opt.Option {
+					if o.Option() == dns.EDNS0EDE {
+						return *o.(*dns.EDNS0_EDE)
+					}
+				}
+			}
+
+			return dns.EDNS0_EDE{}
```

Comment:
Can this use `util.GetEdns0Option`?

```suggestion
		extractEdeOption := func(res *Response) dns.EDNS0_EDE {
			return utils.GetEdns0Option[dns.EDNS0_EDE](res.Res)
```

Reply:
Correct 👍

Missed it during refactoring 🤦‍♂️